proposals/bot-controller-slingshot.md

```diff
@@ -0,0 +1,165 @@
+# Dealbot Controller for Slingshot Retrieval
+
+Authors: @mgoelzer
+
+Initial PR: [#96](https://github.com/protocol/web3-dev-team/pull/96)
+
+<!--
+This template is for a proposal/brief/pitch for a significant project to be undertaken by a Web3 Dev project team.
+The goal of project proposals is to help us decide which work to take on, which things are more valuable than other things.
+-->
+<!--
+A proposal should contain enough detail for others to understand how this project contributes to our team’s mission of product-market fit
+for our unified stack of protocols, what is included in scope of the project, where to get started if a project team were to take this on,
+and any other information relevant for prioritizing this project against others.
+It does not need to describe the work in much detail. Most technical design and planning would take place after a proposal is adopted.
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). 
+Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+Like [#87](https://github.com/protocol/web3-dev-team/pull/87), this proposal is for a Dealbot controller.  Controllers drive the execution of Dealbots for specific purposes.
+
+In order to judge the Slingshot competition, a large number of retrievals need to be made to check the data stored in the competition.  Without this controller, this checking will be substantially more time-consuming and manual.
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+ - The [Dealbots project](https://github.com/protocol/web3-dev-team/pull/84) must resulting in working dealbots that can be run in parallel with tasks dispatched by a single controller.
+ - `{miner,CID}` tuples of Slingshot data must be available.
+ - A low-latency, high-speed link to a go-ipfs node must be available from the bots in the mainnet dealbot cluster.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+
+Input:
+
+ - A list of `{miner,CID}` tuples to be retrieved
```

Comment:
how do you imagine inputting this?

Reply:
I'm imaging a text file like a CSV.  But I think @ribasushi can give us another format like json (any format within reason).

Here's the only catch.  We won't get the entire list at once.  It will be a new list every few days since a human is the one finding the CIDs.  @willscott telll me if you see any problem with that.

Reply:
seems okay